core/impl/src/main/java/com/blazebit/persistence/impl/function/datetime/millisecond/DB2MillisecondFunction.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datetime.millisecond;
+
+/**
+ *
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.4.0
+ */
+public class DB2MillisecondFunction extends MillisecondFunction {
+
+    public DB2MillisecondFunction() {
+        super("MICROSECOND(?1)");
```

Comment:
Shouldn't you also divide the result by 1000 to get milliseconds?

Reply:
Ah yes, I got confused by the DB2 doc. 😄 Now I am reading it again, I see that it indeed has to be divided by 1000.

I'll finish up the PR by adding some tests. I also intend to add date_trunc functions which I quite heavily use.

Reply:
Sounds great, I'll wait for these changes then 👍